conan/cli/formatters/graph/info_graph_html.py

```diff
@@ -57,7 +57,11 @@
         </div>
 
         <script type="text/javascript">
-            const graph_data = {{ deps_graph }};
+            const replaceEscapedCharacters =
+                    (inputString) => inputString.replace(/&#(\d+);/g,
+                    (m, g) => String.fromCharCode(parseInt(g, 10)));
+            const graph_data_raw = "{{ deps_graph }}";
+            const graph_data = JSON.parse(replaceEscapedCharacters(graph_data_raw));
```

Comment:
Isn't there a more "native" way for un-escaping characters? The manual "replace" seems a bit custom, this should be a common thing in javascript?

Reply:
In JavaScript, there are native functions for handling escaped URIs such as `decodeURI` and `decodeURIComponent`, and there's also another deprecated function `unescape`. All of these functions are focused on encoding and decoding URIs and URLs with the pattern "**%XY**".

In our case, we need to escape HTML, so the pattern is "**&#XY;**". In JavaScript, there isn't a native function that converts escaped HTML to a string. This makes sense because typically this step isn't necessary since it's the browser's responsibility to correctly represent these characters.

Our case is special because we have text within HTML that we'll later want to interpret ourselves, rather than simply displaying it on the web.